plugins/org.eclipse.osee.orcs.db/src/org/eclipse/osee/orcs/db/internal/health/PurgeUnusedBackingDataAndTransactions.java

```diff
@@ -156,6 +203,116 @@ public int[] purge() {
          //counts[i++] = purgeAddressedButNonexistentGammas(connection, "osee_txs_archived");
          //counts[i++] = purgeEmptyTransactions(connection);
       }
+      Date date = new Date();
+      SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMdd_hhmmss");
+      String fileName = serverPath + "\\insertStatements_" + dateFormat.format(date) + ".zip";
+      File file = new File(fileName);
+      try {
+         file.createNewFile();
+
+         FileOutputStream fop = new FileOutputStream(file);
+         ZipOutputStream zipOut = new ZipOutputStream(fop);
+
+         try (Writer writer = new OutputStreamWriter(zipOut)) {
+            PrintWriter textWriter = new PrintWriter(writer);
+            addFileToZip(zipOut, OseeDb.ARTIFACT_TABLE.getName(), insertStatements, textWriter);
+            addFileToZip(zipOut, OseeDb.ATTRIBUTE_TABLE.getName(), insertStatements, textWriter);
+            addFileToZip(zipOut, OseeDb.RELATION_TABLE.getName(), insertStatements, textWriter);
+            addFileToZip(zipOut, OseeDb.RELATION_TABLE2.getName(), insertStatements, textWriter);
+            addFileToZip(zipOut, OseeDb.OSEE_SEARCH_TAGS_TABLE.getName(), insertStatements, textWriter);
+            zipOut.putNextEntry(new ZipEntry("ServerResourcesToDelete.txt"));
+            for (String string : uriResourcesToDelete) {
+               textWriter.println(string);
+            }
+            textWriter.flush();
+            zipOut.closeEntry();
+         }
+         fop.flush();
+         fop.close();
+      } catch (IOException ex) {
+         throw OseeCoreException.wrap(ex);
+      }
       return counts;
    }
+
+   public int[] purgeListOfGammas(List<Long> gammasToPurge, List<String> additionalStatements, String filePrefix) {
+      String serverPath = System.getProperty(OseeClient.OSEE_APPLICATION_SERVER_DATA);
+      if (serverPath == null) {
+         serverPath = System.getProperty("user.home");
+      }
+      if (serverPath.equals("null")) {
+         return null;
+      }
+      Path purgeFolder = Paths.get(serverPath + "\\purge");
+      if (Files.exists(purgeFolder)) {
+         serverPath = serverPath + "\\purge";
+      }
+      int i = 0;
+      int[] counts = new int[11];
```

Comment:
I think this is a cut/paste size of 11 from above. It looks like it should be 5? Or should there be a list of Strings that it is the sizeOf that could be iterated over?

Reply:
You are right!  Fixed it : )